src/core/cpu_profiler.cc

```diff
@@ -144,17 +145,22 @@ void cpu_profiler::on_signal() {
     arm_timer(next);
 }
  
-void cpu_profiler::results(std::vector<cpu_profiler_trace>& results_buffer) {
+size_t cpu_profiler::results(std::vector<cpu_profiler_trace>& results_buffer) {
     // Since is this not called in the interrupt it should always succeed
     // in acquiring the lock.
     auto guard_opt = _traces_mutex.try_lock();
     if (!guard_opt.has_value()) {
         results_buffer.clear();
-        return;
+        return 0;
     }
 
     results_buffer.assign(_traces.cbegin(), _traces.cend());
     _traces.clear();
+    
+    auto dropped_samples = _dropped_samples;
+    _dropped_samples = 0;
+
+    return dropped_samples;
```

Comment:
For future consideration: one pattern to replace these three lines is `return std::exchange(_dropped_samples, 0);`